src/dc_receive_imf.rs

```diff
@@ -214,9 +225,15 @@ pub(crate) async fn dc_receive_imf_inner(
         contact::update_last_seen(context, from_id, sent_timestamp).await?;
     }
 
+    let received_msg = match received_msg {
+        Some(m) => m,
+        None => return Ok(None),
```

Comment:
Why is this early exit added? Looks like you don't need the `received_msg` for anything but wrapping it back into `Some` at the end of the function. Why not keep it as is (wrapped in `Option`) and only extract a `chat_id` from it.

If `add_parts` returned `None`, set `chat_id` to `DC_CHAT_ID_TRASH`. But overall I think it's a bug that `add_parts` early exits at all, because when it does, all the code below that creates jobs to move/delete messages on IMAP is skipped. `add_parts` should always create a database record, even if it goes into the trash chat, because otherwise messages are not moved out of inbox and not deleted even if deleting messages on server is enabled. Making `add_parts` not early exit is something for another PR, but this early exit in `dc_receive_imf` should not be added IMO.

Reply:
I'll remove the new early-exit :+1:

> Making add_parts not early exit is something for another PR

I'll create a follow-up issue, then.